pyccel/parser/semantic.py

```diff
@@ -1530,14 +1561,24 @@ def _assign_GeneratorComprehension(self, lhs_name, expr, **settings):
     #====================================================
 
 
-    def _visit(self, expr, **settings):
-        """Annotates the AST.
+    def _visit(self, expr):
+        """
+        Annotate the AST.
 
         The annotation is done by finding the appropriate function _visit_X
         for the object expr. X is the type of the object expr. If this function
         does not exist then the method resolution order is used to search for
         other compatible _visit_X functions. If none are found then an error is
-        raised
+        raised.
+        
+        Parameters
+        ----------
+        expr : pyccel.ast.basic.Basic
+            Object to visit.
+        
+        Returns
+        -------
+        pyccel.ast.variable.Variable
```

Comment:
Missing description of what is returned

Reply:
What do you think of "AST object of type X." as description ?

Reply:
I don't think this is an accurate description. The argument (`expr`) has type `X` but the result sometimes has a different type (e.g. `FunctionCall(PyccelSymbol('shape'), PyccelSymbol('x'))` would return `NumpyShape(Variable('x'))`

Reply:
What would be a better description, should I describe it as just "AST object" ?